src/proof/nova.rs

```diff
@@ -289,62 +308,46 @@ mod tests {
             }
         }
 
-        if check_constraint_systems {
-            let frames = nova_prover.get_evaluation_frames(expr, e, &mut s, limit);
-
-            let multiframes = MultiFrame::from_frames(nova_prover.chunk_frame_count(), &frames, &s);
-            let cs = nova_prover.outer_synthesize(&multiframes).unwrap();
-
-            let adjusted_iterations = nova_prover.expected_total_iterations(expected_iterations);
-            let output = cs[cs.len() - 1].0.output.unwrap();
-
-            if !debug {
```

Comment:
This conditional is actually useful for debugging. For example, when a constraint system is not satisfied, it's useful to be able to set a limit so the test runs only to the point of failure. But without ability to disable these checks, this will lead to a test failure before checking for satisfaction.

Looking at it now, maybe this can be addressed by just moving this whole block down below the `constraint_systems_verified` check. That looks like it might solve the problem, but there might be some other subtlety related to the debugging workflow that I'm forgetting just now.

Reply:
Also the call to `check_cs_deltas`, I think.

Reply:
I moved the code as suggested.